hs_core/management/commands/check_unknown_funder_names_published_resources.py

```diff
@@ -0,0 +1,93 @@
+        parser.add_argument('resource_ids', nargs='*', type=str)
+
+    def handle(self, *args, **options):
+        requests.packages.urllib3.disable_warnings()    # turn off SSL warnings
+
+        def check_funder_name(funder_name):
+            """Checks if the funder name exits in Crossref funders registry.
+            Crossref API Documentation: https://api.crossref.org/swagger-ui/index.html#/Funders/get_funders
+            """
+
+            # match all words in the funder name
+            query = "+".join(funder_name.split())
+            # if we can't find a match in first 50 search records then we are not going to find a match
+            max_record_count = 50
+            email = settings.DEFAULT_SUPPORT_EMAIL
+            url = f"https://api.crossref.org/funders?query={query}&rows={max_record_count}&mailto={email}"
+            funder_name = funder_name.lower()
+            response = requests.get(url, verify=False)
+            found_match = False
+            if response.status_code == 200:
+                response_json = response.json()
+                if response_json['status'] == 'ok':
+                    items = response_json['message']['items']
+                    for item in items:
+                        if item['name'].lower() == funder_name:
+                            found_match = True
+                            return found_match
+                        for alt_name in item['alt-names']:
+                            if alt_name.lower() == funder_name:
+                                found_match = True
+                                return found_match
+                    return found_match
+                return found_match
+            else:
+                msg = "Failed to check funder_name: {} from Crossref funders registry. " \
+                      "Status code: {}".format(funder_name, response.status_code)
+                print(msg)
+                return found_match
+
+        if len(options['resource_ids']) > 0:
+            res_count = len(options['resource_ids'])
+            print(f"TOTAL RESOURCES TO CHECK FOR FUNDER NAMES: {res_count}")
+            print("=" * 100)
+            for res_id in options['resource_ids']:
+                try:
+                    resource = get_resource_by_shortkey(res_id, or_404=False)
+                    if not resource.raccess.published:
+                        print(f"Resource:{res_id} is not a published resource")
+                        continue
+                except ObjectDoesNotExist:
+                    print("No Resource found for id {}".format(res_id))
+                    continue
+                funders = resource.metadata.funding_agencies.all()
+                all_funder_names_matched = True
+                for funder in funders:
+                    if not check_funder_name(funder.agency_name):
+                        print(f"Resource:{res_id} has unknown funder name: {funder.agency_name}")
```

Comment:
Should we include the emails of the resource owners here? I guess it depends on how many unmatched names we anticipate.

I guess we probably will have LOTS of unmatched funder names, so maybe we need to add an arg to this management command, to print out a less verbose option that just has `owner emails`, `resource id`, `unmatched funder name` -- I'm thinking that way we can mail merge and bulk send email notifications out to the owners.

I guess after we run this on Beta tomorrow, we will have a better idea of the best approach

Reply:
I will be making changes to include the resource owners in the output.